pkg/api/processapi/processapi.go

```diff
@@ -139,8 +139,11 @@ type MsgCapabilities struct {
 }
 
 type Binary struct {
-	PathLength int64
+	PathLength int32
+	Reversed   uint32
 	Path       [BINARY_PATH_MAX_LEN]byte
+	End        [BINARY_PATH_MAX_LEN]byte
+	End_r      [BINARY_PATH_MAX_LEN]byte
```

Comment:
For End and End_r swap BINARY_PATH_MAX_LEN for selectors.StringPostfixMaxLength or a new const = 128